swan/Dockerfile

```diff
@@ -0,0 +1,162 @@
+# Installs the extensions and packages required by SWAN
+# to adapt the base alma9 notebook image to SWAN.
+
+# FROM gitlab-registry.cern.ch/swan/docker-images/notebook
+FROM gitlab-registry.cern.ch/pesteves/imageregistry:notebook
+
+LABEL maintainer="swan-admins@cern.ch"
+ARG NB_UID="1000"
+ARG BUILD_TAG=daily
+ENV VERSION_DOCKER_IMAGE=$BUILD_TAG
+
+RUN echo "Building systemuser image with tag ${VERSION_DOCKER_IMAGE}"
+
+# Switch to superuser to install packages
+USER root
+
+# Install required dnf packages
+RUN dnf install -y  git \
+   # gcc \
+   # gcc-c++ \
+   # kernel-devel \
+   # libcurl-openssl-devel \  -- Does not exist in dnf repos
+   # libffi-devel \
+   # ncurses-devel \
+    nano \
+    nodejs \
+    patch \
+   # sqlite-devel \
+    unzip \
+    which \
+   # zeromq3-devel \ -- Does not exist in dnf repos
+    perl-Digest-MD5 \
+    texlive \
+    fontconfig
+
+# Install HEP_OSlibs
+# This installs system packages for LCGs to work
+RUN dnf -y install https://linuxsoft.cern.ch/wlcg/el9/x86_64/wlcg-repo-1.0.0-1.el9.noarch.rpm
+RUN dnf -y install HEP_OSlibs-9.1.0-2.el9
+
+# TODO: Move spark related packages to the third image
+
+# Install required packages
+RUN dnf install -y \
+    # Required by ROOT
+    libXft \
+    # Required by matplotlib
+    tk \
+    # Useful to monitor resource consumption
+    htop \
+    # Install Cloudera dependencies - required by IT Spark clusters
+    alsa-lib \
+    at \
+    bc \
+    cronie \
+    cvs \
+    # db4-cxx \ -- Is not present in DNF repos.
+    # db4-devel \ -- Is not present in DNF repos.
+    file \
+    gdbm-devel \
+    gettext \
+    jpackage-utils \
+    libXtst \
+    man \
+    passwd \
+    # pax \ -- Is not present in DNF repos.
+    # perl-CGI \
+    # perl-ExtUtils-MakeMaker \
+    # perl-Test-Simple \
+    # perl-devel \
+    rsyslog \
+    time \
+    xz-lzma-compat \
+    # Required by Geant4 (libXm)
+    openmotif \
+    # Required by Oracle
+    # TODO: Move to third image: libaio \
+    # Required for key4hep
+    environment-modules && \
+    # Clear the dnf cache as we no longer need to install packages
+    dnf clean all && \
+    rm -rf /var/cache/dnf
+
+# Switch back to jovyan to avoid accidental container runs as root
+USER ${NB_UID}
+
+# Install python pip packages
+RUN mamba install --yes \
+    'jupyter_nbextensions_configurator==0.6.1' \
+    'voila==0.5.5' \
+    'ipyparallel==8.6.1' \
+    # Extensions required by jupyter server
```

Comment:
Should it be in the first image then?

Reply:
Decision is to install `jupyter-server-proxy` (last version is 4.1.0) and this will pull the dependencies below.